src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,347 @@
+        fw.close();
+    }
+
+    private static void lineDecipher(String lineData) {
+        String[] parts = lineData.split("\\|");
+        switch(parts[0].trim()) {
+        case "T":
+            if(parts.length<3) {
+                System.out.println("Invalid Todo task!");
+                break;
+            }
+            tasks.add(new Todo(parts[2].trim()));
+            if(parts[1].trim().equals("1")) {
+                tasks.get(taskCount).markAsDone();
+            }
+            taskCount++;
+            break;
+        case "D":
+            if(parts.length<4) {
+                System.out.println("Invalid Deadline task!");
+                break;
+            }
+            tasks.add(new Deadline(parts[2].trim(), parts[3].trim()));
+            if(parts[1].trim().equals("1")) {
+                tasks.get(taskCount).markAsDone();
+            }
+            taskCount++;
+            break;
+        case "E":
+            if(parts.length<4) {
+                System.out.println("Invalid Event task!");
+                break;
+            }
+            tasks.add(new Event(parts[2].trim(), parts[3].trim()));
+            if(parts[1].trim().equals("1")) {
+                tasks.get(taskCount).markAsDone();
+            }
+            taskCount++;
+            break;
+        default:
+            System.out.println("INVALID TASK DETECTED!");
+            break;
+        }
+    }
+
+    private static String listToString() {
+        String listString = "";
+        for(int i = 0; i<taskCount; i++) {
+            listString = listString.concat(tasks.get(i).toString()+System.lineSeparator());
+        }
+        return listString;
+    }
+
+    private static void inputDecider() throws IncompleteCommandException, UnknownCommandException {
+            //Split command into 2 separate strings "command" and "details"
+            charIndex = userInput.indexOf(' ');
+            if(charIndex ==-1){ // Catch for single word input eg "list" or incomplete command
+                command = userInput;
+                if(command.equals("todo") || command.equals("deadline") || command.equals("event")){
+                    throw new IncompleteCommandException();
+                }
+            } else {
+                command = userInput.substring(0, charIndex).trim();
+                details = userInput.substring(charIndex +1).trim();
+            }
+
+            switch(command){
+            case "list":
+                list();
+                break;
+            case "done":
+                done();
+                break;
+            case "todo":
+                todo();
+                break;
+            case "deadline":
+                deadline();
+                break;
+            case "event":
+                event();
+                break;
+
+            case "delete":
+                delete();
+                break;
+            default:
+                throw new UnknownCommandException();
+            }
```

Comment:
You can actually consider moving Task-related stuff to separate classes/package (similar to the example in week 6 lecture).
Then these  command words can be made to constants in those classes